kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveSessionStoreIntegrationTest.java

```diff
@@ -0,0 +1,300 @@
+    // Given:
+    final Duration inactivityGap = Duration.ofSeconds(5);
+    final Duration gracePeriod = Duration.ofSeconds(2);
+    final SessionWindows window =
+        SessionWindows.ofInactivityGapAndGrace(inactivityGap, gracePeriod);
+
+    final Materialized<String, String, SessionStore<Bytes, byte[]>> store =
+        ResponsiveStores.sessionMaterialized(
+            ResponsiveSessionParams.session(name, inactivityGap, gracePeriod)
+        );
+
+    // Start from timestamp of 0L to get predictable results
+    final List<KeyValueTimestamp<String, String>> inputEvents = asList(
+        new KeyValueTimestamp<>("key", "a", 0L),
+        new KeyValueTimestamp<>("key", "b", 1_000L),
+        new KeyValueTimestamp<>("key", "c", 5_000L)
+    );
+    final List<KeyValue<Windowed<String>, String>> expectedPeeks = List.of(
+        new KeyValue<>(windowedKey("key", 0, 0), "a"),
+        new KeyValue<>(windowedKey("key", 0, 0), null),
+        new KeyValue<>(windowedKey("key", 0, 1000), "ab"),
+        new KeyValue<>(windowedKey("key", 0, 0), null),
+        new KeyValue<>(windowedKey("key", 0, 1000), null),
+        new KeyValue<>(windowedKey("key", 0, 5000), "abc")
+    );
+    final CountDownLatch outputLatch = new CountDownLatch(expectedPeeks.size());
+    final List<KeyValue<Windowed<String>, String>> actualPeeks = new ArrayList<>();
+
+    final StreamsBuilder builder = new StreamsBuilder();
+    final KStream<String, String> input = builder.stream(inputTopic());
+    input
+        .groupByKey()
+        .windowedBy(window)
+        .aggregate(() -> "", sessionAggregator(), sessionMerger(), store)
+        .toStream()
+        .peek((k, v) -> actualPeeks.add(new KeyValue<>(k, v)))
+        .peek((k, v) -> outputLatch.countDown())
+        .selectKey((k, v) -> k.key())
+        .to(outputTopic());
+
+    // When:
+    final Map<String, Object> properties = getMutablePropertiesWithStringSerdes();
+    properties.put(STATESTORE_CACHE_MAX_BYTES_CONFIG, 0);
+    properties.put(APPLICATION_SERVER_CONFIG, "host1:1024");
+    final KafkaProducer<String, String> producer = new KafkaProducer<>(properties);
+    try (
+        final ResponsiveKafkaStreams kafkaStreams =
+            new ResponsiveKafkaStreams(builder.build(), properties);
+    ) {
+      startAppAndAwaitRunning(Duration.ofSeconds(15), kafkaStreams);
+      pipeRecords(producer, inputTopic(), inputEvents);
+
+      assertThat(outputLatch.await(3_000, TimeUnit.MILLISECONDS), Matchers.equalTo(true));
+
+      assertThat(actualPeeks, Matchers.hasSize(expectedPeeks.size()));
+      for (var i = 0; i < actualPeeks.size(); i++) {
+        var actual = actualPeeks.get(i);
+        var expected = expectedPeeks.get(i);
+        assertThat(actual.key, Matchers.equalTo(expected.key));
+        assertThat(actual.value, Matchers.equalTo(expected.value));
+      }
+    }
+  }
+
+  @Test
+  public void shouldComputeSessionAggregateWithMultipleKeys() throws Exception {
```

Comment:
any reason not to just merge this with the first test? (e.g. does it use different functionality)

Reply:
Just to build off of this -- given the startup time for our integration tests, it often makes sense to throw a bit of "good testing practice" out the window and try to do as much as possible within a single test. If it's possible to combine tests without sacrificing the coverage, we should do so.